src/v/storage/log_manager.cc

```diff
@@ -211,23 +219,28 @@ log_manager::housekeeping_scan(model::timestamp collection_threshold) {
      *   compaction is already sequential when this will be unified with
      *   compaction, the whole task could be made concurrent
      */
-    for (auto& log_meta : _logs_list) {
-        co_await log_meta.handle->apply_segment_ms();
-    }
+    while (!_logs_list.empty()
+           && is_not_set(_logs_list.front().flags, bflags::lifetime_checked)) {
+        if (_abort_source.abort_requested()) {
+            co_return;
+        }
 
-    for (auto& log_meta : _logs_list) {
-        log_meta.flags &= ~bflags::compacted;
+        auto& current_log = _logs_list.front();
+        _logs_list.shift_forward();
+
+        current_log.flags |= bflags::lifetime_checked;
+        co_await current_log.handle->apply_segment_ms();
```

Comment:
Had a chat with Andrea and we figured out that this is safe because `apply_segment_ms` grabs a gate that is closed in `log_manager::remove()`.

nit: please add a comment about it

Reply:
done, thanks